### PR TITLE
Remove numeric-id and entity-type from EntityId

### PR DESCRIPTION
**Pull request by adrianheine**
See bug T93172.

### CODE REVIEW

File: src/EntityId.js [hunk 1]
```diff
@@ -107,16 +65,15 @@ wb.datamodel.EntityId = util.inherit(
...
+		return this._serialization;
 	},
 
 	/**
 	 * @inheritdoc
 	 */
 	toJSON: function() {
 		return {
-			'entity-type': this._entityType,
-			'numeric-id': this._numericId
```
Comment (thiemowmde) [inline src/EntityId.js]: Wouldn't it be better to keep this for a while for backwards compatibility?

File: README.md [hunk 2]
```diff
@@ -11,6 +11,15 @@ at the heart of the [Wikibase software](http://wikiba.se/).
 ### 3.0.0 (alpha)
 
 * Removed cloning from `MultiTerm.getTexts`.
+* Turned EntityId into a simple wrapper around an opaque serialization string
+  * Removed EntityId::getNumericId
+  * Removed EntityId::getEntityType
+  * Removed EntityId::getPrefixedId
+  * Introduced EntityId::getSerialization
```
Comment (thiemowmde) [inline README.md]: The `::` is PHP specific. This is JavaScript. ;-)

Comment (adrianheine) [inline README.md]: Not really:
* [Scope resolution operator in C++](https://en.wikipedia.org/wiki/Scope_resolution_operator#C.2B.2B)
* [Scope resolution operator in Ruby](https://en.wikipedia.org/wiki/Scope_resolution_operator#Ruby)
* [Quick access to a classes prototype in CoffeeScript](http://coffeescript.org/#classes)

File: src/EntityId.js (hunk 1 above)
Comment (adrianheine) [inline src/EntityId.js]: I was considering that, too, but it would be difficult since this data is not present anymore.

**Comment by JonasKress** [comment]
+1

**Comment by JonasKress** [comment]
-1 breaks browser tests

**Comment by adrianheine** [comment]
@JonasKress What? I suppose you meant https://gerrit.wikimedia.org/r/#/c/285630/, since this change is in a component that has nothing to do with browser tests. Of course checking out the Wikibase change or just running WikibaseRepo against this change breaks the browser tests, since T132592 is not done yet and so the backend still uses the old serialization. But that has nothing to do with this change.

File: README.md (hunk 2 above)
Comment (thiemowmde) [inline README.md]: I'm sorry, but I don't understand. In JavaScript change logs we write `EntityId.getNumericId` with a single dot.

**Comment by JonasKress** [comment]
Sorry, my fault!

File: README.md (hunk 2 above)
Comment (adrianheine) [inline README.md]: That would be a static class method. This is about a instance methods. You should either do `::` or `.prototype`.

Comment (thiemowmde) [inline README.md]: I find the use of `::` in JavaScript very confusing. When I write `EntityId.getNumericId` I'm describing a specific method in a specific class and do not say if it's static or not.

Comment (adrianheine) [inline README.md]: You definitely should say whether a method is static or not. It's a huge difference.

**Pull request merged**